core/wallet/did.go

```diff
@@ -72,17 +73,40 @@ func (w *Wallet) IsDIDExist(did string) bool {
 }
 
 func (w *Wallet) AddDIDPeerMap(did string, peerID string) error {
+	lastChar := string(did[len(did)-1])
 	var dm DIDPeerMap
 	err := w.s.Read(DIDPeerStorage, &dm, "did=?", did)
 	if err != nil {
 		dm.DID = did
 		dm.PeerID = peerID
+		dm.DIDChar = lastChar
 		return w.s.Write(DIDPeerStorage, &dm)
 	}
+
 	dm.PeerID = peerID
 	return w.s.Update(DIDPeerStorage, &dm, "did=?", did)
 }
 
+func (w *Wallet) AddDIDChar() error {
+	var existingDIDPeer []DIDPeerMap
+	err := w.s.Read(DIDPeerStorage, &existingDIDPeer, "did!=?", "")
+	if err != nil {
+		w.log.Error("Unable to read from DID Peer table", err)
+		return err
+	}
+	for i := 0; i < len(existingDIDPeer); i++ {
```

Comment:
This is to update existing data in DIDPeerStorage table,  right?

So, we need a check on do the query in a way that existing records are not updated every time